tests/run-make/core-ffi-typecheck-clang/rmake.rs

```diff
@@ -0,0 +1,194 @@
+            .iter()
+            .find(|&&(from, _)| from == *llvm_target)
+            .map(|&(_, to)| to)
+            .unwrap_or(llvm_target);
+
+        // Run Clang's preprocessor for the relevant target, printing default macro definitions.
+        let clang_output =
+            clang().args(&["-E", "-dM", "-x", "c", "/dev/null", "-target", &target_to_use]).run();
+
+        let defines = String::from_utf8(clang_output.stdout()).expect("Invalid UTF-8");
+
+        let minicore_content = rfs::read_to_string(&minicore_path);
+        let mut rmake_content = format!(
+            r#"
+            #![no_std]
+            #![no_core]
+            #![feature(link_cfg)]
+            #![allow(unused)]
+            #![crate_type = "rlib"]
+
+            /* begin minicore content */
+            {minicore_content}
+            /* end minicore content */
+
+            #[path = "processed_mod.rs"]
+            mod ffi;
+            #[path = "tests.rs"]
+            mod tests;
+            "#
+        );
+
+        rmake_content.push_str(&format!(
+            "
+            const CLANG_C_CHAR_SIZE: usize = {};
+            const CLANG_C_CHAR_SIGNED: bool = {};
+            const CLANG_C_SHORT_SIZE: usize = {};
+            const CLANG_C_INT_SIZE: usize = {};
+            const CLANG_C_LONG_SIZE: usize = {};
+            const CLANG_C_LONGLONG_SIZE: usize = {};
+            const CLANG_C_FLOAT_SIZE: usize = {};
+            const CLANG_C_DOUBLE_SIZE: usize = {};
+            const CLANG_C_SIZE_T_SIZE: usize = {};
+            const CLANG_C_PTRDIFF_T_SIZE: usize = {};
+            ",
+            parse_size(&defines, "CHAR"),
+            char_is_signed(&defines),
+            parse_size(&defines, "SHORT"),
+            parse_size(&defines, "INT"),
+            parse_size(&defines, "LONG"),
+            parse_size(&defines, "LONG_LONG"),
+            parse_size(&defines, "FLOAT"),
+            parse_size(&defines, "DOUBLE"),
+            parse_size(&defines, "SIZE_T"),
+            parse_size(&defines, "PTRDIFF_T"),
+        ));
+
+        // Generate a target-specific rmake file.
+        // If type misalignments occur,
+        // generated rmake file name used to identify the failing target.
+        let file_name = format!("{}_rmake.rs", target.replace("-", "_").replace(".", "_"));
+
+        // Attempt to build the test file for the relevant target.
+        // Tests use constant evaluation, so running is not necessary.
+        rfs::write(&file_name, rmake_content);
+        let rustc_output = rustc()
+            .arg("-Zunstable-options")
+            .arg("--emit=metadata")
+            .arg("--target")
+            .arg(target)
+            .arg("-o-")
+            .arg(&file_name)
+            .run();
+        rfs::remove_file(&file_name);
+        if !rustc_output.status().success() {
+            panic!("Failed for target {}", target);
+        }
+    }
+
+    // Cleanup
+    rfs::remove_file("processed_mod.rs");
```

Comment:
Nit: not necessary, if a run-make test is successful and you only created temporary files under CWD, the whole directory will be cleaned up.